lib/putter.js

```diff
@@ -55,19 +55,21 @@ Putter.prototype.decode = function (stream) {
 };
 
 /**
- * Knox client error/response callback
+ * Knox client error, response, callback
  */
-Putter.prototype.handle = function (err, res) {
+Putter.prototype.handle = function (err, res, cb) {
     if (err) return this.emit('error', err);
     if(200 == res.statusCode) {
+        if (cb) cb(res.req);
         this.emit('response', { path: res.req.url });
     } else {
+        if (cb) cb('failed status code');
         this.emit('error', new Error('Could not save attachment file, status code: ' + res.statusCode));
     }
 };
 
-Putter.prototype.put = function (string, path, mime, permission) {
-    if (!string || !path || !mime || !permission) 
+Putter.prototype.put = function (string, path, mime, permission, cb) {
```

Comment:
It'd be cool if after this line we did
```
Meteor.bindEnvironment(cb)
```
once we have tests we can see if this works.